### ISSUE TITLE
Fix how we collect tax forms

### ISSUE THREAD

**Issue by alanna** [opencollective/opencollective#1774]
Our current approach to collecting tax forms (W9 and W8-BEN/E) is broken. We missed getting these forms from a significant number of people in 2018, despite investing a lot of time and effort. We need to improve for 2019.

**Why this is important:**

Open Collective risks not being in IRS compliance, and our users will have trouble filing their own taxes if we don't issue them a 1099.

**Requirements:**

* Users who invoice over >$600 must provide us a tax form (this does not apply for receipt reimbursements - [more details here](https://docs.opencollective.com/help/expenses/tax-information)).
* We need a way to clearly determine if they owe us a form, and have road blocks in place so we don't pay their expense if they have not submitted the tax form when required.

**Current process:**

* Tax form bot makes a comment on all expenses that push a user over the $600 threshold (regardless of if it's an invoice or receipt)
* The user gets this notification and follows the instructions
* They select the correct form and email it to w9@opencollective.com
* Alanna manually downloads these and uploads them to the tax form dropbox folders
* If someone over the threshold does not have a tax form on file, we hold off paying their expenses
* At the end of the year, we do a big data export of all users who have been paid >$600, and Alanna manually goes through them to see if we have their form
* Alanna manually emails the people who we should have a form for but don't
* Some reply with the form, some do not
* The accountants to their best to issue the required 1099s but many fall through the cracks 

**Problems:**

* System does not differentiate between invoice and receipt expenses, but only invoice payments count toward the $600 limit
* Tax form bot only posts on one expense, but users might have many expenses, so it's not a reliable signal for the host admin to look out for the tax bot's comment
* Checking if someone has a form on file is a totally manual process, prone to error
* Sometimes people have different names attached to their tax ID, user account, and email account, including pseudonyms, company names, multiple personal names, etc—very hard to confirm if we have the form on file to match the Open Collective user account
* Sometimes people are paid even when their form has not been submitted
* If they have already been paid, we have no leverage to get them to respond to our request for the tax form
* Users are often confused about which form they need to file and how to answer some of the questions (especially W8 BEN-E)
* At the end of the year, we don't have all the forms we need to be in IRS compliance

**Ideas for solutions**

* Make the system differentiate between invoices and expenses so we can separate that data
* Clearly flag if a user has hit the threshold and if we have their tax form on file - disallow paying out their expense if they don't have the right flag
* Have the user to submit the form through Open Collective, automatically associated with their user account, so we don't have to match up their tax name and their user name, which are often different

**Potential workflow**

* When user submits an expense (not on expense approval, see #1305) that puts them over the $600 threshold, they get a warning message saying they need to submit a tax form and that we've sent them an email with details
* A red flag is put on their account, disabling payouts for invoice expenses (but not receipt reimbursements)
* They get an email with details about the required tax form, and a button that says "upload your form"
* That button takes them to a page on Open Collective where they submit the form
* The host admin is notified and they check the form and approve it if it's valid
* User gets a green flag and their expense can be paid
* At the end of the year, we export all the required tax form for the accountants at once

**Related issues:**

* #1386 shows screenshots of a third party service we could use
* #1167 includes a design that would seperate out invoices from expense reimbursements

**Comment by piamancini** [comment]
We've been kicking this (and our very old expense flow) down the road for too long. It's boring and it doesn't bring more money, but it's important.

**Comment by alannallama** [comment]
Added another example of a third party system for automating this (from HelloSign) in #1386 

@znarf - would it be possible for you to look into this and estimate how much work it would take to integrate? 

I think we could potentially do a manual-assisted process using the free version, if we don't want to invest in integration.

https://www.hellosign.com/products/helloworks

**Comment by alannallama** [comment]
@Betree this is the big hairy tax forms issue I mentioned to you. It would be great if a dev could look into scope/feasibility of integrating a third party solution like the ones mentioned here, or we could work together on another solution.

**Comment by alanna** [comment]
@pietgeursen I head back from the accountant - we need to collect a new tax form each year from each person who hits the threshold for that year.

**Comment by pietgeursen** [comment]
Ok good to know!

**Comment by alanna** [comment]
Update after speaking with a couple other US-based fiscal hosts: They really only pay out receipt reimbursements so this feature won't be relevant to them. 

One definitely would not appreciate the system automatically requesting tax forms from people who go over $600 in receipt reimbursements but don't actually owe them a form, and when they do need to collect a W9 they have their own systems (integrated via QuickBooks across all their payables not just OC payouts).

So, in conclusion, we should just build this for Open Source Collective for now.

**Comment by pietgeursen** [comment]
I've been studying this and I think Alanna's proposal is on point.

It sounds like this feature must only be available for Open Source Collective.

I've taken Alanna's proposal and tweaked it slightly:

**MVP**

* When user submits any invoice that takes their total for the year over $600 (just ignore the expense case for now.) then: 
* A red flag is put on their account
* They get an email with details about the required tax form, and a button that says "upload your form"
* That button takes them to a page on Open Collective where they submit the form
* The flag is visible in the host dashboard and if the host tries to approve an expense, they get a warning that they can override.
* The host admin is notified and they check the form and approve it if it's valid
* There's somewhere for the host to go and set the flag to 'approved' for that user. (could be forest or somewhere sensible in the ui)
* At the end of the year, we export all the required tax form for the accountants at once

## What about HelloWorks

Hello works is kinda like google forms, if the results of the form input could then be automatically added to a pdf template.

This _seemed_ really promising: We could make a beautiful flow that asks just the right questions, depending if they're a US person etc and then it generates a nice pdf at the end. They even have a template for the W9. But we would need to make our own W8BEN. I looked at their example code for their W9. It's [1700 lines](https://gist.github.com/pietgeursen/d2374f1bb706803f0bd52d2c26d69ff4) and not really worth it to try and learn whatever they're doing. I don't think how to do this is even documented.

I don't think this is an option.

## What about HelloSign

Advantages:

- We might get better engagement / higher response rates because filling out forms on HellosSign is really easy.
- We could automate some things:
  - When the user crosses the threshold, we hit the HelloSign api. They get an email.
  - When they've signed, HelloSign hits our server with a link to the signed document. It can also email the host at the same time.
  - We could probably remove the manual validation step by the host as HelloSign does form validations for you. 
- Probably faster to build?
- There's an official hellosign node module that might save a little time.
- Cost is actually minimal, $13 a month.

Disadvantages:

- Vendor lock in?

Now that I've written this, I actually think we should use HelloSign.

**Comment by alanna** [comment]
Thanks for that @pietgeursen 

Helloworks has a form builder UI, so I could build the forms and workflows, no code required. If we think there's a benefit to proving Helloworks' nice flow experience, form building should not be the blocker. That way, we could send everyone to the same flow from our app and they answer questions and work their way through a decision tree providing the info we need.

I'm open to using HelloSign instead. It would be quite a different experience, where they have to figure out for themselves what form to use, and then fill it out in the ugly and confusing IRS PDF formats, but at least all online. We do get a fair few people writing to us confused about which form to use and how to fill it out (especially between W8-BEN and W8 BEN-E), which could be lessened by walking them through it step by step, but I don't think that's the most important problem we're trying to solve.

Another option would be to just build it ourselves. It's actually a really simple workflow that we could accomplish with a form on our site natively. It could ask you a couple questions, then give you the link to the form to download, fill out, and sign, and a button to upload the signed version. If we wanted to avoid vendor lock-in, this would be the way to go. Even if it were really ugly and basic, it would be 100x better than our current system.

The most important things to solve is making sure we ask for forms from people who need to submit them, provide them a functional way to do that, and mark in our system if they have one on file so we don't pay them without it. Either Helloworks or HelloSign, or a DIY solution, could accomplish this.

@pietgeursen should I have a go at designing the DIY form option, so you can see how complex or simple it is from a programming perspective? Or do you reckon a 3rd party system is the way to go.

**Comment by pietgeursen** [comment]
@alanna 

> Helloworks has a form builder UI, so I could build the forms and workflows, no code required. 

Yeah, I saw the form builder when I was looking at their example w9. In the form builder they had a warning for the component that displays one path or another depending on the selection. It said something along the lines of "this component must be used with the code builder." And then there were 1700 lines of code.

If we can build what we need only in the form builder that would be great. That might mean re-implementing the w9 rather than just using their version.

We need to check if HelloWorks supports the ability to populate either the w8 or the w9 pdf based on user input. Because it might assume you're only trying to populate one pdf.

>  Or do you reckon a 3rd party system is the way to go.

At this point yes.

**Comment by Betree** [comment]
Specs seem good, would be nice indeed to have an estimation of how much time it would take to integrate with HelloWorks or HelloSign VS homemade.

**Comment by alanna** [comment]
@pietgeursen and I are moving forward with HelloWorks for the moment, because I figured out how to create the workflow we need without code. There are a lot of details to it so it will take me some time to complete, but so far testing is going well. We're also researching HelloSign if it turns out HelloWorks doesn't do what we need.

**Comment by pietgeursen** [comment]
I've created a [project)(https://github.com/opencollective/opencollective/projects/21) that breaks up the tasks into smaller pieces.

I'd appreciate @Betree  having a look over the tickets I created in case there are better ways to do things.

**Comment by alanna** [comment]
I have completed a draft of the full tax form workflow, and it's ready for testing. There are likely still small tweaks to be made. You can launch it [here](https://portal.helloworks.com/link/nrKucJBFEIfDagUK). Feel free to fill it out - this doesn't send the data anywhere yet, just captures it in HelloWorks.

Testing  notes:

* This form flow collects the data for W9, W8BEN and W8BENE depending on the options you select.
* When testing, decide on a test case (ex. overseas company or US individual) and go start to finish. Certain options trigger other questions to appear, but unselecting those options doesn't cause those extra questions to disappear, so if you go back and change earlier answers you'll be left with irrelevant questions at the end.
* This form covers the use case of every tax form we've ever had submitted. However, form W8BENE is also used by heaps of other types of entities we don't interact with and including them would be hugely complex. I have stated in the form that if you don't fit one of the three common classification options (active NFFE, passive NFFE, or nonprofit), not to submit the form and to contact us. I don't think this will ever actually come up, but that's why it only lists 3 out of like 30 possible FATCA classifications.

@pietgeursen @piamancini please test if you get a chance.

**Comment by piamancini** [comment]
great! We are starting the expense flow wireframes process soon, so will be great to have this tested before and incorporate. We should also add a flag to enable users to differentiate between invoices and reimbursements.

**Comment by Betree** [comment]
@pietgeursen Regarding the cards:

> Create a hook that gets triggered when the user's invoice total for the year crosses $600 for an enabled host.

A daily (or hourly) job would be appropriate. This is a process that can be delayed and run asynchronously without slowing down the main thread.

> Add a new table to the db that can track the tax form for a user account

This one seems good, we'll be able to easily extend that if we need to collect other documents in the future. I'd decouple that from the notion of tax ; something like a `LegalDocuments` table (and `document_link` instead of `tax_form_link`) and a `type` column could bring more genericity at a little cost (eg. type=W9, type=IDENTITY_PAPERS,type=COMPANY_CERTIFICATE...etc).

Example: if a host require collectives admins to provide their IDs, we can easily extend the table just by using a different `type`.

I wonder if we shouldn't user `collective_id` instead of `user_id`, that we we could collect documents for collectives too. @znarf you recently advocated that we should do that for applications, does that stand here too?

> Should we rip out the Tax Bot?

Regarding the fact that tax bot publishes public comments on expenses, it won't be able to show private links to the user. We should probably replace it by an email, or a notification bar that stays on top of the page until the form is submitted if we really want to push it agressively :slightly_smiling_face:

**Comment by piamancini** [comment]
Should we rip out the Tax Bot?
Once this is done, yes.

**Comment by pietgeursen** [comment]
> something like a LegalDocuments table 

Good idea. Have used that.

> and a type column could bring more genericity at a little cost

Also used that too.

> I wonder if we shouldn't user collective_id instead of user_id

Yep, good call. Did that too.

Thanks @Betree

**Comment by xdamman** [comment]
This is what I could find about good practice and compliance to store SSN (social security numbers): https://security.stackexchange.com/questions/143235/what-is-the-best-practice-industry-standard-for-storing-documents-with-social

Amazon S3 is PCI compliant. But as a good practice, we should encrypt the documents that we store and keep the private key away from S3.

**Comment by pietgeursen** [comment]
Ok so, [PR for the main chunk of work](https://github.com/opencollective/opencollective-api/pull/2074) is up for review. 

When it is merged:
- @alanna and I will work to manually update the db for people who have already submitted tax forms for this year.
- This [PR for the frontend](https://github.com/opencollective/opencollective-frontend/pull/1987) will be able to flag expenses that need a tax form.
- Once a few other details (s3 buckets, pdf encryption) have been sorted, then [this PR](https://github.com/opencollective/opencollective-api/pull/2145) can be merged which will turn on the cron job so that users get sent an email from hello works.
